### ISSUE TITLE
Plugin for a `getopts` like utility.

### ISSUE THREAD

**Issue by bucaran** [bpinto/oh-my-fish#329]
Every utility / function needs to handle arguments. What usually happens is that every function does all the grunt work to parse `$argv` itself, and while some patterns do tend to recur, it's different almost every time. What do you think about a fishy [`getopts`](http://en.wikipedia.org/wiki/Getopts)?

+ [`getopts` reference](http://pubs.opengroup.org/onlinepubs/9699919799/utilities/getopts.html)
+ [Utility Syntax Guidelines](http://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap12.html#tag_12_02).

```fish
function my_utility
  while set option (get.opts "x y z a:a-long-option w:with-arg:" $argv)
    switch $option
      case x
      case y
      ...
      case w
      ...
      case \*
    end
  end
end
```

**Comment by bpinto** [comment]
Something similar is already being done on fish-shell, I think we should help them instead: https://github.com/fish-shell/fish-shell/issues/478

**Comment by bucaran** [comment]
Sure. It looks complicated though and it's probably going to take a while. On the other hand I already have a decent solution (at least better than parsing stuff every time) we can start using right now, `get`, a simple plugin anyone can use in the meantime.

**Comment by bpinto** [comment]
Let's see it. :)

I find `get` a weird name for this plugin though. I think we should keep the original name, `getopts`.
```
getopts.parse
```

**Comment by bucaran** [comment]
Thought you would say so!

Mmm, `getopts` does not really work either because it does one thing `getopts` does not. It returns _operands_ instead of modifying a cryptic global `OPTIND` you would have to check yourself to see if the array is within bounds, so you can save a check and get right to your operands right away.

There is already the default `argv`, so `get.args`, is another option, what do you think?

**Comment by bpinto** [comment]
Well, it's an improvement on top of `getops`, but other than that, it behaves the same, doesn't it?
Even if it's not a replacement, I find it a good name.

P.S.: I would prefer `getoptions`. :)

**Comment by bucaran** [comment]
From the point of view of the user it would work roughly the same as you could see by inspecting the snippet above, internally I use different global variables and make sure to delete them after completion we are finished, so the user can just concentrate on getting its options and not unsetting environment variables with cryptic names, etc.

**Comment by bpinto** [comment]
Yeah, seems like `getopts` is the best name. It doesn't really matter how it behaves internally, as long as it has the same usage, I guess.

**Comment by bucaran** [comment]
Cool. Will PR when it's ready and we can finish the discussion then.

**Comment by bpinto** [comment]
Closed on #339
